js/views/settingsVw.js

```diff
@@ -0,0 +1,290 @@
+      timezone_option.attr("selected",user.time_zone == t.offset);
+      timezone_str += timezone_option[0].outerHTML;
+    });
+
+    __.each(languageList, function(l, i){
+        var language_option = $('<option value="'+l.langCode+'">'+l.langName+'</option>');
+        language_option.attr("selected",user.language == l.langCode);
+        language_str += language_option[0].outerHTML;
+    });
+
+    ship_country.html(ship_country_str);
+    currency.html(currency_str);
+    country.html(country_str);
+    timezone.html(timezone_str);
+    language.html(language_str);
+  },
+
+  validateInput: function(e) {
+    "use strict";
+    e.target.checkValidity();
+    $(e.target).closest('.flexRow').addClass('formChecked');
+  },
+
+  tabClick: function(activeTab, showContent){
+    "use strict";
+    this.$el.find('.js-tab').removeClass('active');
+    activeTab.addClass('active');
+    this.$el.find('.js-tabTarg').addClass('hide');
+    showContent.removeClass('hide');
+  },
+
+  generalClick: function(e){
+    this.tabClick($(e.target).closest('.js-tab'), this.$el.find('.js-general'));
+  },
+
+  shippingClick: function(e){
+    this.tabClick($(e.target).closest('.js-tab'), this.$el.find('.js-shipping'));
+  },
+
+  storeClick: function(e){
+    this.tabClick($(e.target).closest('.js-tab'), this.$el.find('.js-store'));
+  },
+
+  blockedClick: function(e){
+    this.tabClick($(e.target).closest('.js-tab'), this.$el.find('.js-blocked'));
+  },
+
+  advancedClick: function(e){
+    this.tabClick($(e.target).closest('.js-tab'), this.$el.find('.js-advanced'));
+  },
+
+  cancelClick: function(e){
+      Backbone.history.loadUrl();
+  },
+
+  saveClick: function(e){
+        var self = this;
+        var server = self.options.userModel.get('server_url');
+
+        var settings_form = this.$el.find("#settingsForm");
+        settings_form.addClass('formChecked');
+        if(!settings_form[0].checkValidity()) {
+            self.showErrorModal("Errors in form", "Please fix all errors in the form attempting to save again");
+            return;
+        }
+
+        //As there are 3 different API urls we need to call,
+        //we need to split up our form data into 3 parts,
+        //depending on which API call each value belongs to
+        var profileFormData = new FormData();
+        var settingsFormData = new FormData();
+        var uploadImageFormData = new FormData();
+        $.each(settings_form.find("input,select,textarea"),
+            function(i,el) {
+                var id = $(el).attr("id");
+                if(id == "country") {
+                    profileFormData.append("location",$(el).val());
+                }
+                if(id == "name" || id == "handle") {
+                    profileFormData.append(id,$(el).val());
+                } else if(id == "avatar") {
+                    uploadImageFormData.append(id,$(el)[0].files[0]);
+                } else if($(el).attr("type") == "checkbox") {
```

Comment:
This is a little dangerous, since it assumes only one of the APIs will receive data from a checkbox. You don't have to change it, I'm just noting this for the record.